src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,146 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.ColumnBuilder;
+import edu.washington.escience.myria.column.builder.ColumnFactory;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleUtils;
+
+/**
+ * Generic apply operator.
+ */
+public class Split extends UnaryOperator {
+  /***/
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Name of column to split on regex. Must have String type.
+   */
+  private final String columnName;
+
+  /**
+   * Compiled regex to split input tuples.
+   */
+  private final Pattern pattern;
+
+  /**
+   * Schema of output tuples.
+   */
+  private Schema outputSchema;
+
+  /**
+   * Index of column to split in input schema (and anonymous result column in output schema).
+   */
+  private int splitColIdx;
+
+  /**
+   * Array of column indices in output schema corresponding to column indices of input schema. The column to be split
+   * has its index mapped to the sentinel value -1.
+   */
+  private int[] inputColIndexToOutputColIndex;
+
+  /**
+   * 
+   * @param child child operator that data is fetched from
+   * @param columnName name of string column to split using regex
+   * @param regex regular expression to split string input column on
+   */
+  public Split(final Operator child, final String columnName, final String regex) {
+    super(child);
+    this.columnName = columnName;
+    pattern = Pattern.compile(regex);
+  }
+
+  @Override
+  protected TupleBatch fetchNextReady() throws DbException {
+    final Operator child = getChild();
+    for (TupleBatch tb = child.nextReady(); tb != null; tb = child.nextReady()) {
+      List<ColumnBuilder<?>> colBuilders = new ArrayList<>(tb.numColumns());
```

Comment:
Can you look at `TupleBatchBuilder` and see if it satisfies your needs? I would have expected you to use it...